internal/internal_utils.go

```diff
@@ -38,23 +38,45 @@ import (
 	"golang.org/x/net/context"
 )
 
-// versionHeaderName refers to the name of the
-// tchannel / http header that contains the client
-// library version
-const versionHeaderName = "cadence-client-version"
+const (
+	// libraryVersionHeaderName refers to the name of the
+	// tchannel / http header that contains the client
+	// library version
+	libraryVersionHeaderName = "cadence-client-library-version"
+
+	// featureVersionHeaderName refers to the name of the
+	// tchannel / http header that contains the client
+	// feature version
+	featureVersionHeaderName = "cadence-client-feature-version"
+
+	// languageHeaderName refers to the name of the
+	// tchannel / http header that contains the client
+	// language
+	languageHeaderName  = "cadence-client-language"
```

Comment:
I would rename it to cadence-client-name
and value be something like uber-go, uber-java, as someone else can create mylibrary-java, etc.

Reply:
May be even call it user-agent to match HTTP?

Reply:
changed to be "cadence-client-name" : "uber-go"